File: sdk/samples/all in on code/Visual Studio 2008/CSWinFormDesigner/MouseEvents/UC_DetectMouseEvent.cs
/************************************* Module Header **************************************\
* Module Name:	UC_DetectMouseEvent.cs
* Project:		CSWinFormDesigner
* Copyright (c) Microsoft Corporation.
* 
* 
* The DetectMouseEvent sample demonstrates how to detect mouse events at design time.
* 
* This source is subject to the Microsoft Public License.
* See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
* All other rights reserved.
* 
* History:
* * 4/25/2009 3:00 PM Zhi-Xin Ye Created
* * 
* 
\******************************************************************************************/


using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Windows.Forms.Design;

namespace CSWinFormDesigner.MouseEvents
{
    [Designer(typeof(UC_DetectMouseEventDesigner))]
    public partial class UC_DetectMouseEvent : UserControl
    {
        public UC_DetectMouseEvent()
        {
            InitializeComponent();
        }
    }

    class UC_DetectMouseEventDesigner : ControlDesigner
    {
        protected override bool GetHitTest(Point point)
        {
            MessageBox.Show("Mouse at: " + point.ToString());

            return base.GetHitTest(point);
        }

        protected override void OnMouseEnter()
        {
            // Do something here when the mouse enters the control at design time;
            base.OnMouseEnter();
        }

        protected override void OnMouseLeave()
        {
            // Do something here when the mouse leaves the control at design time;
            base.OnMouseLeave();
        }
    }
}

File: sdk/samples/all in on code/Visual Studio 2008/VBWinFormDesigner/MouseEvents/UC_DetectMouseEvent.vb
'************************************* Module Header **************************************\
' Module Name:	UC_DetectMouseEvent.vb
' Project:		VBWinFormDesigner
' Copyright (c) Microsoft Corporation.
' 
' 
' The DetectMouseEvent sample demonstrates how to detect mouse events at design time.
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
' All other rights reserved.

' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'***************************************************************************'

Imports System.Windows.Forms.Design
Imports System.ComponentModel

Namespace MouseEvents
    <Designer(GetType(UC_DetectMouseEventDesigner))> _
    Public Class UC_DetectMouseEvent

    End Class


    Friend Class UC_DetectMouseEventDesigner
        Inherits ControlDesigner

        Protected Overrides Function GetHitTest(ByVal point As Point) As Boolean
            MessageBox.Show(("Mouse at: " & point.ToString))
            Return MyBase.GetHitTest(point)
        End Function

        Protected Overrides Sub OnMouseEnter()
            MyBase.OnMouseEnter()
        End Sub

        Protected Overrides Sub OnMouseLeave()
            MyBase.OnMouseLeave()
        End Sub

    End Class
End Namespace